samples/msal-node-samples/ExpressTestApp/TestApp/App/utils/cachePlugin.js

```diff
@@ -1,38 +1,58 @@
-const fs = require("fs");
-const cachePath = './App/data/cache.json' // replace this string with the path to your valid cache file.
+/**
+* A custom cache plugin for persisting the msal's token cache, using a persistence solution like Redis or MongoDB
+* This effectively implements the ICachePlugin interface from msal. For more information, visit:
+* https://azuread.github.io/microsoft-authentication-library-for-js/ref/interfaces/_azure_msal_common.icacheplugin.html
+*/
 
-const beforeCacheAccess = async (cacheContext) => {
-    return new Promise(async (resolve, reject) => {
-        if (fs.existsSync(cachePath)) {
-            fs.readFile(cachePath, "utf-8", (err, data) => {
-                if (err) {
-                    reject();
+module.exports = (persistenceHelper, sessionId) => {
+    return {
+        beforeCacheAccess: async (cacheContext) => {
+            return new Promise(async (resolve, reject) => {
+                
+                // express session ids start with the string "sess:"
+                persistenceHelper.get("sess:" + sessionId, (err, sessionData) => {
+                    if (err) {
+                        console.log(err);
+                        reject();
+                    }
+
+                    if (sessionData) {
+                        persistenceHelper.get(JSON.parse(sessionData).account.homeAccountId, (err, cacheData) => {
+                            if (err) {
+                                console.log(err);
+                                reject();
+                            }
+                            cacheContext.tokenCache.deserialize(cacheData);
+                            resolve();
+                        });
+                    } else {
+                        resolve();
+                        return;
+                    }
+                });
+            });
+        },
+        afterCacheAccess: async (cacheContext) => {
+            return new Promise((resolve, reject) => {
+
+                if (cacheContext.cacheHasChanged) {
+                    const kvStore = cacheContext.tokenCache.getKVStore();
+
+                    // getting homeAccountId from account entity in kvStore
+                    const homeAccountId = Object.values(kvStore)[1]["homeAccountId"];
```

Comment:
We should validate that `kvStore` even has any values and then validate that `homeAccountId` is defined before moving on to the next part